### PR TITLE
feat(icon-switch): design updates

### LINKED ISSUE #1536: [Icon content switcher] Styling enhancement
Looking to update the icon content switcher styling slightly to match what Carbon has for their [content switcher](https://react.carbondesignsystem.com/?path=/story/contentswitcher--content-switcher-playground)

Updates:
- Update color of hover and selected states to match Carbon's content switcher
- remove stroke around default state (although that is in design on the PAL site but I'm not seeing it in code?)
- Round corners on the left and right end buttons
- Build in light variant
- Can we also update the placement of the tooltip that shows up to match what the [icon button](https://react.carbondesignsystem.com/?path=/story/button--playground) does?

Carbon style specs: https://www.carbondesignsystem.com/components/content-switcher/style

Specs:

Icon content switcher

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #1536 

**Summary**

- Bring IconSwitch into alignment with new styles. 

[Icon content switcher - Style.pdf](https://github.com/carbon-design-system/carbon-addons-iot-react/files/7139752/Icon.content.switcher.-.Style.pdf)

**Change List (commits, features, bugs, etc)**

- update styles
- add disabled/children props to IconSwitch
- update stories to include more icons and show the divider
- add new stories

**Acceptance Test (how to verify the PR)**

- confirm styles match those shown in the PDF above

**Regression Test (how to make sure this PR doesn't break old functionality)**

- confirm this doesn't break interactions or designs in other stories that use the switchers. 
- DashboardEditor
- ImageGalleryModal
- HotspotTextStyleTab

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by bjornalm** [comment]
When I select the different buttons the icons inside move slightly. 
click-2-573cb55b7f0e

**Comment by bjornalm** [comment]
The corners look a bit weird here in the dark button. 
image
It is in RTL mode for https://deploy-preview-2819--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-imagegallerymodal--basic

**Comment by kevinsperrine** [comment]
> The corners look a bit weird here in the dark button.
> image
> It is in RTL mode for https://deploy-preview-2819--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-imagegallerymodal--basic
> 
> Same story and setup, possibly related, the selection also looks strange:
> image

This one is fixed.

**Comment by kevinsperrine** [comment]
@bjornalm Should all be covered now.

**Comment by bjornalm** [comment]
Are the latest changes in the preview deployment? I'm still seeing some strange things.

Selection in RTL still looks off:
image

Hovering shows the divider:
image

The selection is only shown on top and bottom
image


Perhaps we can look at how Carbon solved it for the content switcher? In a smaller  container/window it looks like our iconSwitch
https://react.carbondesignsystem.com/?path=/story/components-contentswitcher--default&knob-Selection%20mode%20(selectionMode)=automatic&knob-[Deprecated]:%20Light%20variant%20(light)=true

**Comment by kevinsperrine** [comment]
> Are the latest changes in the preview deployment? I'm still seeing some strange things.
> 
> Selection in RTL still looks off:
> image
> 
> Hovering shows the divider:
> image
> 
> The selection is only shown on top and bottom
> image
> 
> Perhaps we can look at how Carbon solved it for the content switcher? In a smaller container/window it looks like our iconSwitch
> [https://react.carbondesignsystem.com/?path=/story/components-contentswitcher--default&knob-Selection%20mode%20(selectionMode)=automatic&knob-[Deprecated]:%20Light%20variant%20(light)=true](https://react.carbondesignsystem.com/?path=/story/components-contentswitcher--default&knob-Selection%20mode%20(selectionMode)=automatic&knob-%5BDeprecated%5D:%20Light%20variant%20(light)=true)

Ugh. This design more complicated than it should be. fixed those issues. just waiting for them to push and deploy.

**Comment by davidicus** [comment]
@kevinsperrine Doesn't the spec call for rounded borders even when focused?

**Comment by kevinsperrine** [comment]
> @kevinsperrine Doesn't the spec call for rounded borders even when focused?

Yeah, that was part of the problem. I had accidentally used 1px radius instead of the 0.25rem radius, so that's why it looked funky.

**Comment by davidicus** [comment]
Internal buttons outline looks a little light. I think I saw that you are overwriting the outline prop and doing something with gradients. Perhaps we can scope those to `:not(first-child), :not(last-child)`?
image

**Review by davidicus — APPROVED** [review]

**Comment by kevinsperrine** [comment]
> Internal buttons outline looks a little light. I think I saw that you are overwriting the outline prop and doing something with gradients. Perhaps we can scope those to `:not(first-child), :not(last-child)`? I think this looks pretty good but thought I mention it.
> image

By light, so you mean the top/bottom or left/right? I can thicken the L/R if that's what you mean...

**Comment by bjornalm** [comment]
This is looking much better!

One last small thing is that the divider is still showing between a selected or hovered button a normal state button. It is really nitty gritty but if we want this component to be pixel perfect we should probably address it. The divider makes the line look a bit "fuzzy":

image

and if we disable the component it becomes even more visible:

image

**Comment by kevinsperrine** [comment]
> This is looking much better!
> 
> One last small thing is that the divider is still showing between a selected or hovered button a normal state button. It is really nitty gritty but if we want this component to be pixel perfect we should probably address it. The divider makes the line look a bit "fuzzy":
> 
> image
> 
> and if we disable the component it becomes more visible and on both sides:
> 
> image

Are we done yet? 😉   https://deploy-preview-2819--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-iconswitch--example-used-in-content-switcher

**Comment by bjornalm** [comment]
> > This is looking much better!
> > One last small thing is that the divider is still showing between a selected or hovered button a normal state button. It is really nitty gritty but if we want this component to be pixel perfect we should probably address it. The divider makes the line look a bit "fuzzy":
> > image
> > and if we disable the component it becomes more visible and on both sides:
> > image
> 
> Are we done yet? 😉 https://deploy-preview-2819--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-iconswitch--example-used-in-content-switcher

Hehe, I have actually found more stuff, but I won't torture you with that. Nothing that will break a build.

File: packages/react/src/components/IconSwitch/_icon-switch.scss [hunk 1]
```diff
@@ -6,72 +6,95 @@ $icon-switch-size-large: $spacing-09;
...
+  }
+
+  &:hover {
+    // hide the divider on hover
+    &:not(.#{$iot-prefix}--icon-switch--light) {
+      .#{$iot-prefix}--icon-switch__divider:after {
+        display: none;
+      }
+
+      // hide the divider beside it on hover to prevent "fuzziness" of the hover state
```
Comment (bjornalm) [inline packages/react/src/components/IconSwitch/_icon-switch.scss:31]: hehe

**Review by bjornalm — APPROVED** [review]
Great work, you have earned a pixel-perfect badge!

File: packages/react/src/components/IconSwitch/_icon-switch.scss (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/components/IconSwitch/_icon-switch.scss:31]: :-) There are so many random little rules in there, I figured it good to label some of them.

**Comment by kevinsperrine** [comment]
@davidicus This one is ready to merge now, but coveralls is down, so we'll either need to wait for that or force merge. https://status.coveralls.io/

**Pull request merged**